src/spikeinterface/preprocessing/filter.py

```diff
@@ -178,7 +183,9 @@ class BandpassFilterRecording(FilterRecording):
         Margin in ms on border to avoid border effect
     dtype : dtype or None
         The dtype of the returned traces. If None, the dtype of the parent recording is used
-    {}
+    **filter_kwargs : dict
```

Comment:
You mean the `{}` is used for injecting? Do you have any more details - I can't find anything obvious when Googling. At the moment the `{}` appears in the API: [BandpassFilterRecording](https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.preprocessing.bandpass_filter)

Reply:
I think this is actually missing!!

this should be added after the function definition at the bottom (similar to [this](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/base.py#L839)):
```
filter.__doc__ = filter.__doc__.format(_common_filter_docs)
bandpass_filter.__doc__ = bandpass_filter.__doc__.format(_common_filter_docs)
notch_filter.__doc__ = notch_filter.__doc__.format(_common_filter_docs)
highpass_filter.__doc__ = highpass_filter.__doc__.format(_common_filter_docs)
```

Reply:
Note that this docstring hacking does not work with editors like vscode that rely on static analysis:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/2a41ed53-d95d-4eca-ade0-3ff86ef5fae8)

but do work on notebooks with the `?` operator.

Reply:
Yeah I've just memorized those parts of the docstrings for my purposes :)

Reply:
Once the {} is added in and the `.format` then this PR looks good to me!

Reply:
@chrishalcrow can you add those lines?

Reply:
Hello - lots of debate online about how to include **kwargs in docstrings! Numpy and SciPy doesn't document any individual kwargs (https://numpy.org/doc/1.26/reference/generated/numpy.arctan.html, https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html) but we are only including a subset of 3 of them, so I think we should document them.

At the moment, things are whitespaced so that `**filter_kwargs` is a numpydoc Parameter. Then the actual filter kwargs (filter_order, filter_mode, ftype) are just part of the description. I think this is correct: the description tells you how to write the filter_kwargs dict. Does the whitespace look ok here:

<img width="664" alt="Screenshot 2024-06-18 at 18 56 26" src="https://github.com/SpikeInterface/spikeinterface/assets/57948917/bd14c030-80c5-482d-a93b-9bf96b3c9eb3">
?

Also, .format needs to be applied to the functions rather than the classes, and the notch filter doesn't take these, so isn't included.

Reply:
Sorry, just realised that `filter` takes `filter_order`, `filter_mode` and `ftype` as actual arguments rather than in a `**filter_kwargs` dict, so I've updated this. This now passes the numpydoc validation.